Esri/arcgis-maps-sdk-swift-samples
# [Setup] Add favorites to the Sample Viewer

## Description

This PR implements the ability to favorite samples in the Sample Viewer similar to the old iOS Sample Viewer. A "Favorites" category was added, where the user can see a list of their favorited samples. Samples can be added by pressing the plus button in the category or by long pressing on a sample row anywhere in the app. Samples in the list can also be removed and rearranged by tapping the "Edit" button in the category. A few other views were modified to best accommodate the new views. 

## Linked Issue(s)

- `swift/issues/4370`

## How To Test

- In the Favorites category, tap on the plus button to add a sample to favorites.
- Tap on "Edit" to reorder/delete the favorited samples.
- Tap and hold on a sample row anywhere in the Sample Viewer to favorite/unfavorite from the context menu.
- Close and reopen the app to ensure the favorited samples persist.

## Screenshots

|Before|After|
|:-:|:-:|
| ![Simulator Screenshot - iPhone 15 Pro - 2023-12-06 at 16 52 03](https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/34120440/97872de2-fd31-4ec9-8630-e6b6bcdb0e1d) | ![Simulator Screenshot - iPhone 15 Pro - 2023-12-06 at 16 45 06](https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/34120440/87ec7202-5c2e-4bad-b05e-7042d4eaa875) |

### Other: 
 ![Simulator Screenshot - iPhone 15 Pro - 2023-12-06 at 16 45 00](https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/34120440/e32e7e4f-25b4-4a21-bbdc-16f5a201d090)   ![Simulator Screenshot - iPhone 15 Pro - 2023-12-06 at 16 56 58](https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/34120440/8df952ec-bc5a-4de0-9e15-ed2268a67f5b)
